accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+
+
+### Signing methods
+#### Author managed keys
+The most common existing solution to signing software involves maintainer-managed keys, which has known [usability](https://www.usenix.org/legacy/events/sec99/full_papers/whitten/whitten.ps) [and](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html) [security](https://blog.cryptographyengineering.com/2014/08/13/whats-matter-with-pgp/) [issues](https://www.wired.co.uk/article/efail-pgp-vulnerability-outlook-thunderbird-smime). Registries that have implemented public key signing have seen either limited adoption or achieved adoption only by mandating it (e.g., Maven Central, which reports developer friction).
+
+Distributing author-managed public keys (PKI) remains a challenge as well as guaranteeing secure handling of private key material. Leaked credentials are a widespread problem among existing PKI implementations, as evidenced by the large number of SSH keys that both [independent researchers](https://www.ndss-symposium.org/wp-content/uploads/2019/02/ndss2019_04B-3_Meli_paper.pdf) and GitHub’s internal secret scanning program have found over the years.
+
+#### Long-lived X.509 certificates
+Long-lived X.509 certificates are in widespread use to secure web pages, but they are still cumbersome to manage.
+
+Long-lived certificates have the advantage of being able to chain up to a certificate authority that has intrinsic trust, so beyond authenticating that the artifact signed by the certificate is valid, a consumer can validate that the artifact is from an author that they trust.
+
+Like other author-managed keys, the developer must keep and secure the private certificate throughout the lifetime of the certificate. Long-lived certificates may also encounter issues with needing to manage revocation in the event of a private certificate leak. While a centralized certificate authority could issue long-lived X.509 certificates to developers as Let's Encrypt has done for web pages, the need to manage revocation across software artifacts in the registry is a problem.
+
+Nuget, for example, supports [CA-issued code signing certificates](https://docs.microsoft.com/en-us/nuget/create-packages/sign-a-package). Its seen low adoption among the open source community due to several reported [problems](https://haacked.com/archive/2019/04/03/nuget-package-signing/).
+
+#### Short-lived X.509 certificates
+Short-lived X.509 certificates eliminate the need to manage keys across a long lifetime and eliminate the need to handle revocations.
+
+Combined with a trusted certificate authority that can create short-lived certificates and log the signatures that took place on them, we can validate any artifacts that were signed by that certificate, and protect against repudiation by maintaining an immutable log of the signing activities.
+
```

Comment:
It may be worth mentioning that signed timestamps also support enforcing that a certificate was valid when the signature was created.

There is some good documentation around this by Authenticode - https://docs.microsoft.com/en-us/windows/win32/seccrypto/time-stamping-authenticode-signatures

Reply:
Thanks included this 👍